itou/metabase/management/commands/sql/005_suivi_saisies_dans_asp.sql

```diff
@@ -43,11 +42,17 @@ FROM
         left join "fluxIAE_AnnexeFinanciere_v2" af
             on emi.emi_afi_id = af.af_id_annexe_financiere
             /* Ne prendre en compte que les structures qui ont une annexe financière 
-            valide  pour l'année en cours */
+            valide  pour l'année en cours et prendre en compte les structures qui sont en retard de saisie dans l'asp */
             and af_etat_annexe_financiere_code in ('VALIDE', 'SAISI')
-            and date_part('year', to_date(af.af_date_debut_effet, 'dd/mm/yyyy')) = date_part('year', current_date)
-            /* Ne prendre que les déclarations mensuelles de l'année en cours */
-            and  emi.emi_sme_annee = date_part('year', current_date)
+            and date_part('year', af.af_date_debut_effet_v2) >= (date_part('year', current_date) - 1)
+            and af.af_date_fin_effet_v2 
+                    >= 
+                    make_date(
+                        cast (date_part('year', current_date) as integer), 
+                        cast((date_part('month', current_date) - 3) as integer),1
```

Comment:
Que représente le `3` ici ? Il y a une logique de délai de 3 mois quelque part ?

Par ailleurs je suis pas un expert du `CAST` mais tu es sûr qu'il fonctionne comme attendu malgré la double parenthèse ?

Reply:
Je peux remplacer avec `and af.af_date_fin_effet_v2 >= CURRENT_DATE - INTERVAL '3 months'`

Reply:
Ok oui ce serait plus lisible.